src/main/java/com/jwplayer/jwplatform/rest/HttpCalls.java

```diff
@@ -0,0 +1,101 @@
+package com.jwplayer.jwplatform.rest;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.TreeMap;
+
+import org.json.JSONObject;
+
+import com.jwplayer.jwplatform.exception.JWPlatformException;
+import com.jwplayer.jwplatform.exception.JWPlatformUnknownException;
+import com.jwplayer.jwplatform.utils.Util;
+import com.mashape.unirest.http.HttpResponse;
+import com.mashape.unirest.http.JsonNode;
+import com.mashape.unirest.http.Unirest;
+import com.mashape.unirest.http.exceptions.UnirestException;
+
+public class HttpCalls {
+
+	public static JSONObject request(final String path, final Map<String, String> params, final boolean isBodyParams,
+			final String requestType, final Map<String, String> headers) throws JWPlatformException {
+		final String requestUrl;
+		final HttpResponse<JsonNode> response;
+
+		try {
+			switch (requestType.toUpperCase()) {
```

Comment:
Does it make sense to use separate functions rather than this monolith with a switch statement? (i.e. getRequest(), postRequest() etc.)

Reply:
To be honest, given the number of fixed cases here, I don't mind either. I could make the change but it will require test changes everywhere which is okay if we do want to do it. At this point, I don't see a strong reason why.

Reply:
ok, let's leave for now.